doc/doc-support/options-doc.nix

```diff
@@ -0,0 +1,28 @@
+# To build this derivation, run `nix-build -A nixpkgs-manual.optionsDoc`
+{ lib, nixosOptionsDoc }:
+
+let
+  modules = lib.evalModules {
+    modules = [ ../../pkgs/top-level/config.nix ];
+    class = "nixpkgsConfig";
+  };
+
+  root = toString ../..;
+
+  transformDeclaration =
+    decl:
+    let
+      declStr = toString decl;
+      subpath = lib.removePrefix "/" (lib.removePrefix root declStr);
+    in
+    assert lib.hasPrefix root declStr;
+    {
+      url = "https://github.com/NixOS/nixpkgs/blob/master/${subpath}";
+      name = subpath;
+    };
```

Comment:
I noticed you changed this to an `assert` when it was this before: https://github.com/NixOS/nixpkgs/blob/6a037089e1d7f14ac4b1d0f560a45d01b7fd0f8e/doc/default.nix#L89-L93

Notably I'd expect this to break manual rendering when building non-Nixpkgs modules using [`documentation.nixos.includeAllModules`](https://search.nixos.org/options?channel=24.05&show=documentation.nixos.includeAllModules&from=0&size=50&sort=relevance&type=packages&query=AllModules)

Reply:
Yes! I thought that for the nixpkgs manual, there wouldn't be ... non-nixpkgs modules. Am I wrong there?

Reply:
Ah yes you're right! Mixed things up here :)